src/python/pants/backend/python/target_types.py

```diff
@@ -972,7 +972,59 @@ def __post_init__(self):
         assert self.entry_points_field or self.provides_field
 
 
+class WheelField(BoolField):
+    alias = "wheel"
+    default = False
+    help = "Whether to build a wheel for the distribution."
+
+
+class SDistField(BoolField):
+    alias = "sdist"
+    default = False
+    help = "Whether to build an sdist for the distribution."
+
+
+class ConfigSettingsField(DictStringToStringSequenceField):
+    """Values for config_settings (see https://www.python.org/dev/peps/pep-0517/#config-settings).
+
+    NOTE: PEP-517 appears to be ill-defined wrt value types in config_settings. It mentions that:
+
+    - Build backends may assign any semantics they like to this dictionary, i.e., the backend
+      decides what the value types it accepts are.
+
+    - Build frontends should support string values, and may also support other mechanisms
+      (apparently meaning other types).
+
+    Presumably, a well-behaved frontend is supposed to work with any backend, but it cannot
+    do so without knowledge of what types each backend expects in the config_settings values,
+    as it has to set those values.
+
+    See a similar discussion in the context of Pip: https://github.com/pypa/pip/issues/5771 .
+
+    In practice, the backend we currently care about, setuptools.build_meta, expects a
```

Comment:
I think what this really says is we only support `setuptools.build_meta`. This is underscored by the fact this rides on top of a `SetupPyChroot` which is almost certainly the wrong shape for a another builder - for example poetry  - to use.

Reply:
I think that's right, but I'm not fully convinced yet. For example, if we didn't generate `setup.py` but assumed there was one in the target closure, we could equally assume there was a "whatever builder X needs in order to work" in the target closure. 

In other words we could just throw all the code we have at the builder and assume it knows what to do.

Reply:
I think you recently switched the `SetupPyChroot` to not have sources mounted under `src/` so this would happen to work now with poetry - for example - but only just so. You can imagine and sort of setup though for a particular builder and they definitely could require some opinionated layout we don't match. For an existing setup.py though, our `SetupPyChroot` might not work at all - even before this change. Of course that would just mean the person migrating that `setup.py` to Pants would need to make the same adjustments they'd need now.